src/ivoc/ivocmain.cpp

```diff
@@ -218,6 +218,45 @@ void penv() {
 }
 #endif
 
+extern "C" {
+#if DARWIN || defined(__linux__)
+#include <dlfcn.h>
+#include <string>
+
+/* It is definitely now the case on mac and I think sometimes the case on
+linux that dlopen needs a full path to the file. A path to the binary
+is not necessarily sufficent as one may launch python or nrniv on the
+target machine and the lib folder cannot be derived from the location of
+the python executable.This seems to be robust if this file is inside a
+shared library.
+The return value ends with a '/' and if the prefix cannot be determined
+the return value is "".
+*/
+const char* path_prefix_to_libnrniv() {
+  static char* path_prefix_to_libnrniv_ = NULL;
+  if (!path_prefix_to_libnrniv_) {
```

Comment:
I believe (or hope) with CMake this won't be needed as we add rpath.

Reply:
> there are testing aspect that one could tackle if it is easier. 

I don't understand this comment

Reply:
I may not have understood whole context but I was referring to below comment: 

> and I think sometimes the case on  linux that dlopen needs a full path to the file.

I was wondering with RPATH handling logic added in [this cmake](https://github.com/neuronsimulator/nrn/blob/28245d266b524fd186e5938380a64400097f6662/cmake/RpathHelper.cmake#L13) file whether it would be handled.

Reply:
>  RPATH handling logic added in this cmake file

Isn't the problem with $ORIGIN that it refers to the folder of the launched executable? So
if python is launched, there may in general be no information in regard to where libnrnmpi... is? Admittedly, in the wheel context everything is in a single install location so lib is unambiguously relative to bin.